python/ray/serve/_private/http_state.py

```diff
@@ -230,6 +230,20 @@ def shutdown(self):
         self._shutting_down = True
         ray.kill(self.actor_handle, no_restart=True)
 
+    def is_ready_for_shutdown(self) -> bool:
+        """Return whether the HTTP proxy actor is shutdown.
+
+        For an HTTP proxy actor to be considered shutdown, it must be marked as
+        _shutting_down and the actor must be dead.
+        """
+        actor_id = self.actor_handle._actor_id.hex()
+        actor_state = "ALIVE"
+        for _actor in ray._private.state.actors().values():
+            if _actor["ActorID"] == actor_id:
+                actor_state = _actor["State"]
+                break
+        return self._shutting_down and actor_state == "DEAD"
```

Comment:
this is pretty hacky, we shouldn't need to use `ray._private.state.actors()` for normal operations, it's built for observability purposes.

can we instead track this in the normal reconcile loop? the health check ref will return `RayActorError` once the proxy is dead.